cloudify_rest_client/plugins_update.py

```diff
@@ -0,0 +1,132 @@
+
+    @property
+    def execution_id(self):
+        return self['execution_id']
+
+    @property
+    def deployments_to_update(self):
+        return self['deployments_to_update']
+
+    @property
+    def created_at(self):
+        return self['created_at']
+
+
+class PluginsUpdateClient(object):
+    """
+    Cloudify's plugins update management client.
+    """
+
+    def __init__(self, api):
+        self.api = api
+        self._uri_prefix = 'plugins-updates'
+        self._wrapper_cls = PluginsUpdate
+
+    def get(self, plugins_update_id, _include=None, **kwargs):
+        """
+        Gets a plugins update by its id.
+
+        :param plugins_update_id: PluginsUpdate's id to get.
+        :param _include: List of fields to include in response.
+        :return: The plugins update details.
+        """
+        assert plugins_update_id
+        uri = '/{self._uri_prefix}/{id}'.format(
+            self=self, id=plugins_update_id)
+        response = self.api.get(uri, _include=_include, params=kwargs)
+        return self._wrapper_cls(response)
+
+    def _wrap_list(self, response):
+        return ListResponse(
+            [self._wrapper_cls(item) for item in response['items']],
+            response['metadata']
+        )
+
+    def list(self, _include=None, sort=None, is_descending=False, **kwargs):
+        """
+        Returns a list of available plugins updates.
+        :param _include: List of fields to include in response.
+        :param sort: Key for sorting the list.
+        :param is_descending: True for descending order, False for ascending.
+        :param kwargs: Optional filter fields. For a list of available fields
+               see the REST service's models.PluginsUpdate.fields
+        :return: Plugins list.
+        """
+        params = kwargs
+        if sort:
+            params['_sort'] = '-' + sort if is_descending else sort
+
+        response = self.api.get('/{self._uri_prefix}'.format(self=self),
+                                _include=_include,
+                                params=params)
+        return self._wrap_list(response)
+
+    def update_plugins(self, blueprint_id, force=False):
+        """
+        Updates the plugins in all the deployments that use the given
+        blueprint.
+
+        :param blueprint_id: blueprint ID to perform the update with.
+        :param force: if to forcefully update when other non-active plugins
+         updates exists associated with this blueprint.
+        :return: a PluginUpdate object.
+        """
+        params = {
+            'force': force
+        }
+        response = self.api.post(
+            '/{self._uri_prefix}/{}/update/initiate'.format(blueprint_id,
+                                                            self=self),
+            params=params
+        )
+        return PluginsUpdate(response)
+
+    def finalize_plugins_update(self, plugins_update_id):
+        """
+        Finalize the plugins update.
```

Comment:
is internal or for client use?

Reply:
Internal but used by the plugins update workflow

Reply:
also add here a note about this being internal and in the rest-docs and cli docs

Reply:
There's no CLI command for this (neither did deployments update). Updated the docstring